pkg/vpcmodel/grouping.go

```diff
@@ -219,24 +219,44 @@ func (g *groupingConnections) addPublicConnectivity(ep EndpointElem, commonProps
 	(*g)[ep][connKey].appendNode(targetNode)
 }
 
-// vsiGroupingBySubnets returns a slice of EndpointElem objects, by grouping set of elements that
-// represent network interface nodes from the same subnet into a single groupedNetworkInterfaces object
-func vsiGroupingBySubnets(groupedConnLines *GroupConnLines, elemsList []EndpointElem) []EndpointElem {
+// vsiOrSubnetsGroupingBySubnetsOrVsis given *GroupConnLines, a list of EndpointElem and a bool saying whether
+// the EndpointElem represents VSIs or subnets.
+// It returns a slice of EndpointElem objects, by grouping the input set of EndpointElem into the input *GroupConnLines
+// such that
+// 1. If the grouped elements are vsis (as by the input bool) then the elements are grouped by their subnet
```

Comment:
```suggestion
// It returns a slice of EndpointElem objects, by grouping the input set of EndpointElem
// such that
// 1. If the grouped elements are vsis (as by the input bool) then the elements are grouped by their subnet
```

Reply:
`groupedConnLines` is used only for caching.. can be changed in a follow up PR , to send only the relevant cache instead of this entire object. (which is misleading)

Reply:
It will change once we merge with main after we merge the unification PR.